phpBB/phpbb/di/service_collection.php

```diff
@@ -103,4 +103,35 @@ public function get_service_classes()
 	{
 		return $this->service_classes;
 	}
+
+	/**
+	 * Returns the service associated to a class
+	 *
+	 * @return mixed
+	 * @throw \RuntimeException if the
+	 */
+	public function get_by_class($class)
+	{
+		$service_id = null;
+
+		foreach ($this->service_classes as $id => $service_class)
+		{
+			if ($service_class === $class)
+			{
+				if ($service_id !== null)
+				{
+					throw new \RuntimeException('Many services found for class "'.$class.'" in collection.');
```

Comment:
I think `More than one service definitions found for...` would be a bit more descriptive of what the problem really is. Other than this, LGTM.